### PR TITLE
Add enterprise analytics tutorial

### PR DESCRIPTION
**Pull request by smithjilks**
# What type of PR is this?

### CODE REVIEW

**Review by SammyOina — CHANGES_REQUESTED** [review]
include prism screenshots, and cli output as well

File: enterprise-analytics/README.md [hunk 1]
```diff
@@ -0,0 +1,374 @@
+# Enterprise Supply Chain Demand Forecasting
+
+Confidential multi-party analytics for enterprise supply chain optimization. Three competing retail companies collaboratively train a demand forecasting model inside a Trusted Execution Environment (TEE)—each contributes proprietary transaction data, but **no company ever sees another's raw data**.
+
+This example demonstrates:
+
+- **Secure Computation (aTLS)** — Attested TLS verifies the TEE hardware and software stack before any data is uploaded
+- **Multi-Party Computation** — Three independent data providers each upload proprietary datasets into the same encrypted enclave
+- **Real-World Data** — Uses the [UCI Online Retail II](https://www.kaggle.com/datasets/mashlyn/online-retail-ii-uci) dataset (real European e-commerce transactions) split across simulated companies
+- **Enterprise Value** — Benchmark proves the consortium model outperforms any single-company model
```
Comment (fbugarski) [inline enterprise-analytics/README.md:10]: In practice this is not guaranteed. In my local run, the consortium model achieved lower R² than at least one of the individual models. The claim should be softened.

File: enterprise-analytics/README.md [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+
+Dataset preparation complete. 3 company datasets saved to 'datasets/'
+```
+
+## Train Model (Local)
+
+To train the consortium model locally:
+
+```bash
+python train.py
```
Comment (fbugarski) [inline enterprise-analytics/README.md:149]: This step is sensitive to the current working directory. Running train.py from the repository root fails because it expects datasets/ relative to CWD. This is not clearly documented and makes the workflow non-intuitive.

Comment (fbugarski) [inline enterprise-analytics/README.md:149]: This step is sensitive to the current working directory. Running train.py from the repository root fails because it expects datasets/ relative to CWD. This is not clearly documented and makes the workflow non-intuitive.

File: enterprise-analytics/README.md [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+- **aTLS (Attested TLS):** Each participant verifies the TEE's hardware attestation before uploading. The cryptographic quote proves the enclave is genuine AMD SEV-SNP/Intel TDX hardware running the exact agreed-upon algorithm.
+- **Memory encryption:** All data inside the TEE is encrypted by the CPU. The cloud provider, hypervisor, and host OS have zero access.
+- **No raw data exits:** Only aggregated model weights and statistical reports leave the enclave. Individual transaction records are destroyed.
+
+## Setup Virtual Environment
+
+```bash
+python3 -m venv venv
+source venv/bin/activate
+pip install -r requirements.txt
```
Comment (fbugarski) [inline enterprise-analytics/README.md:111]: The example references a requirements.txt, but no such file is present in the repository. This makes the setup incomplete.

File: enterprise-analytics/README.md [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+```
+
+## Install
+
+Fetch the data from Kaggle — [Online Retail II UCI](https://www.kaggle.com/datasets/mashlyn/online-retail-ii-uci) dataset:
```
Comment (fbugarski) [inline enterprise-analytics/README.md:116]: The setup assumes Kaggle CLI + credentials, but does not clearly explain the need for a legacy API key (kaggle.json). This can be confusing with the newer Kaggle token system.

Comment (fbugarski) [inline enterprise-analytics/README.md:116]: The setup assumes Kaggle CLI + API credentials, but it may help to be more explicit about the credential setup required for CLI-based downloads. In practice, the current Kaggle UI exposes multiple token/key options, which can make this step a bit confusing.

File: enterprise-analytics/README.md [hunk 5]
```diff
@@ -0,0 +1,374 @@
+# Enterprise Supply Chain Demand Forecasting
+
+Confidential multi-party analytics for enterprise supply chain optimization. Three competing retail companies collaboratively train a demand forecasting model inside a Trusted Execution Environment (TEE)—each contributes proprietary transaction data, but **no company ever sees another's raw data**.
```
Comment (fbugarski) [inline enterprise-analytics/README.md:3]: These guarantees are not enforced by the example code itself and appear to rely on the underlying platform. It may be worth clarifying that these are platform-level guarantees rather than properties of this example implementation.

File: enterprise-analytics/README.md [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+│                    Trusted Execution Environment (TEE)           │
+│                   AMD SEV-SNP / Intel TDX Hardware               │
+│  ┌───────────────────────────────────────────────────────────┐  │
+│  │                    In-Enclave Agent                        │  │
+│  │                                                           │  │
+│  │  ┌─────────────┐  ┌─────────────┐  ┌─────────────┐      │  │
+│  │  │ Company 1   │  │ Company 2   │  │ Company 3   │      │  │
+│  │  │ Dataset     │  │ Dataset     │  │ Dataset     │      │  │
+│  │  │ (encrypted) │  │ (encrypted) │  │ (encrypted) │      │  │
+│  │  └──────┬──────┘  └──────┬──────┘  └──────┬──────���      │  │
```
Comment (fbugarski) [inline enterprise-analytics/README.md:70]: There seem to be some rendering/encoding issues in the diagrams (e.g. broken characters like `���` in the GitHub view). It may be worth double-checking encoding to ensure the documentation renders correctly.

Comment (fbugarski) [inline enterprise-analytics/README.md:70]: There seem to be some rendering/encoding issues in the diagrams (e.g. broken characters like `���` in the GitHub view). It may be worth double-checking encoding to ensure the documentation renders correctly.

File: enterprise-analytics/train.py [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+Enterprise Supply Chain Demand Forecasting - Training Algorithm
+
+This algorithm runs inside a Trusted Execution Environment (TEE) on the
+Prism AI / Cocos platform. It receives proprietary sales datasets from
+multiple competing retailers and trains a unified demand forecasting model
+without any party seeing another's raw data.
+
+Scenario:
+    Three retail companies contribute their transaction histories to jointly
+    train an XGBoost model that predicts monthly product demand. The combined
```
Comment (fbugarski) [inline enterprise-analytics/train.py:11]: This claim sounds too strong as a general statement. In my local run, at least one individual company model achieved a higher R² than the consortium model, so the consortium does not consistently outperform single-company models. It may be better to phrase this as a potential benefit (e.g. “can capture broader market signals and may improve performance”) rather than a guaranteed outcome.

Comment (fbugarski) [inline enterprise-analytics/train.py:11]: This claim sounds too strong as a general statement. In my local run, at least one individual company model achieved a higher R² than the consortium model, so the consortium does not consistently outperform single-company models. It may be better to phrase this as a potential benefit (e.g. “can capture broader market signals and may improve performance”) rather than a guaranteed outcome.

File: enterprise-analytics/train.py [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+
+    # ── Train consortium model (all companies together) ──────────────────
+    print("\n" + "=" * 60)
+    print("TRAINING CONSORTIUM MODEL (all companies)")
+    print("=" * 60)
+
+    X = combined[FEATURE_COLS].values
+    y = combined[TARGET_COL].values
+
+    X_train, X_test, y_train, y_test = train_test_split(
```
Comment (fbugarski) [inline enterprise-analytics/train.py:201]: Since this is presented as a demand forecasting example, a time-based split would likely be more appropriate than a random train/test split for evaluation. That would better reflect how the model performs on future periods rather than shuffled observations from the same overall time range.

Comment (fbugarski) [inline enterprise-analytics/train.py:201]: Since this is presented as a demand forecasting example, a time-based split would likely be more appropriate than a random train/test split for evaluation. That would better reflect how the model performs on future periods rather than shuffled observations from the same overall time range.

Comment (fbugarski) [inline enterprise-analytics/train.py:201]: Since this is presented as a demand forecasting example, a time-based split would likely be more appropriate than a random train/test split for evaluation. That would better reflect how the model performs on future periods rather than shuffled observations from the same overall time range.

Comment (fbugarski) [inline enterprise-analytics/train.py:201]: Since this is presented as a demand forecasting example, a time-based split would likely be more appropriate than a random train/test split for evaluation. That would better reflect how the model performs on future periods rather than shuffled observations from the same overall time range.

File: enterprise-analytics/train.py [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+    """Create time-series demand features from raw transaction data."""
+    df = df.copy()
+    df["InvoiceDate"] = pd.to_datetime(df["InvoiceDate"])
+    df["YearMonth"] = df["InvoiceDate"].dt.to_period("M")
+    df["Month"] = df["InvoiceDate"].dt.month
+    df["DayOfWeek"] = df["InvoiceDate"].dt.dayofweek
+    df["WeekOfYear"] = df["InvoiceDate"].dt.isocalendar().week.astype(int)
+
+    # Aggregate to monthly product-level demand
+    monthly = (
```
Comment (fbugarski) [inline enterprise-analytics/train.py:56]: The code comment mentions "monthly product-level demand", but including `WeekOfYear` in the grouping means the aggregation is no longer purely monthly. This effectively splits data within the same month, so it may be worth either adjusting the grouping or clarifying the description.

File: enterprise-analytics/predict.py [hunk 10]
```diff
@@ -0,0 +1,173 @@
+"""
+Enterprise Supply Chain Demand Forecasting - Inference / Result Analysis
+
+Loads the trained consortium demand model and produces evaluation metrics,
+visualizations, and a summary report demonstrating the value of multi-party
+collaborative analytics over single-company models.
+"""
+
+import os
+import sys
```
Comment (fbugarski) [inline enterprise-analytics/predict.py:10]: The `sys` import appears to be unused and can likely be removed.

**Review by fbugarski — CHANGES_REQUESTED** [review]

**Review by fbugarski — CHANGES_REQUESTED** [review]
I was able to run the full pipeline locally end-to-end.

Overall, the example works well, but there are a few areas that would benefit from clarification or adjustment:

- some claims around consortium performance are stronger than what is observed in practice
- the local workflow is sensitive to the current working directory (datasets path)
- setup steps (requirements and Kaggle credentials) could be made more explicit
- a few minor inconsistencies between code and documentation (e.g. aggregation wording)
